BSP/STM32F413/Src/BSP_CAN.c

```diff
@@ -2,25 +2,89 @@
 
 #include "BSP_CAN.h"
 #include "stm32f4xx.h"
+#include "os.h"
 
 #define CAN_MODE        CAN_Mode_Normal
 
-static CanTxMsg TxMessage;
-static CanRxMsg RxMessage;
+// Queue functions
 
-static bool RxFlag = false;
+typedef struct _msg {
+    uint32_t id;
+    uint8_t data[8];
+} msg_t;
+
+#define QUEUE_SIZE 10
+typedef struct _queue {
+    msg_t msgs[QUEUE_SIZE];
+    
+    uint8_t head;
+    uint8_t tail;
+} queue_t;
+
+static queue_t gRxQueue = {
+    {0},
+    0,
+    0,
+    true
+};
```

Comment:
This has an extra parameter. I think one of my other PRs fixes this along with some other issues.

Reply:
Fixed by #263